### PR TITLE
Disable editing the "Others" icon in UI (only in advanced mode)

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: #12683

- Cartodb.js PR: https://github.com/CartoDB/cartodb.js/pull/1726
- Disable editing the "Others" icon

For acceptance:

- Create a map with a layers of points
- Style the layer by value, using a string column.
- Check "Others" item.

others-img-disabled

NOTE: using CartoCSS in advanced the "Others" icon can be set (this is a hack) and the legend works as expected.

others-img-hack

### CODE REVIEW

**Comment by Jesus89** [comment]
@rochoa Are you OK with this change (empty url like _url('')_) ?

**Comment by rochoa** [comment]
Not a big fan of this approach. We are relying on a hack, using an empty URL.

Turbo-Carto ramps are not intended for these scenarios. If you want a more granular control over "others", you need to handle that logic in CartoCSS itself and filter the categories already in use to assign the "others" file. An example:

```css
#layer {
  marker-width: 20;
  marker-fill: ramp([sov_a3], (#5F4690, #1D6996, #38A6A5, #0F8554, #73AF48, #EDAD08, #E17C05, #CC503E, #94346E, #6F4070, #666666), ("USA", "RUS", "CHN", "BRA", "CAN", "AUS", "IND", "MEX", "ARG", "IDN"), "=");
  marker-fill-opacity: 1;
  [sov_a3 != "USA"][sov_a3 != "RUS"][sov_a3 != "CHN"][sov_a3 != "BRA"][sov_a3 != "CAN"][sov_a3 != "AUS"][sov_a3 != "IND"][sov_a3 != "MEX"][sov_a3 != "ARG"][sov_a3 != "IDN"] {
    marker-file:  url('https://s3.amazonaws.com/com.cartodb.users-assets.production/maki-icons/park2-18.svg');
  }
  marker-file: ramp([sov_a3], (url('https://s3.amazonaws.com/com.cartodb.users-assets.production/maki-icons/parking-18.svg')), ("USA"), "=");
  marker-allow-overlap: true;
  marker-line-width: 1;
  marker-line-color: #FFF;
  marker-line-opacity: 1;
}
```

**Comment by Jesus89** [comment]
The issue with this method is that the "default" value does not appear in the rules used for the legends:

rules-missing-others

**Comment by Jesus89** [comment]
Ok, these are the conclusions:

- UI can not allow the "Other" image to be set
- But legends are ready to display the "Others" image if it is set in advance mode

So let's update the ticket's name.

Thank you guys!

cc @xavijam @rochoa

**Comment by xavijam** [comment]
Thanks to you majo!

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/categories-list/categories-list-view.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 var CustomListView = require('../../../../../../custom-list/custom-list-view');
 
 module.exports = CustomListView.extend({
-  _renderItem: function (mdl) {
+  _renderItem: function (model, index) {
```
Comment (rubenmoya) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/categories-list/categories-list-view.js]: 😍

**Comment by rubenmoya** [comment]
Looks good!

**Comment by ivanmalagon** [comment]
It's being deployed at ded02 for the next 'accepter'.

**Comment by rubenmoya** [comment]
Working as expected! :shipit:

**Pull request merged**